components/compositing/compositor.rs

```diff
@@ -412,9 +412,7 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
     }
 
     pub fn deinit(self) {
-        if let Err(err) = self.rendering_context.make_gl_context_current() {
-            warn!("Failed to make GL context current: {:?}", err);
-        }
+        self.rendering_context.make_current();
```

Comment:
We should probably allow the `RenderingContext` to propagate failure still so that the `Compositor` can do error handling. Just because it's only warning now doesn't mean it shouldn't handle errors. Quite likely it needs to stop itself and ask the application to restart it when this happens. None of that needs to happen here, but I think the `RenderingContext` should still expose the errors when they do happen.